editor/src/components/inspector/controls/color-picker-swatches.tsx

```diff
@@ -0,0 +1,185 @@
+import * as React from 'react'
+import { Button, colorTheme, FlexColumn, FlexRow } from '../../../uuiui'
+import { v4 as UUID } from 'uuid'
+import { Substores, useEditorState } from '../../editor/store/store-hook'
+import { useDispatch } from '../../editor/store/dispatch-context'
+import { updateColorSwatches } from '../../editor/actions/action-creators'
+import { ColorSwatch } from '../../editor/store/editor-state'
+
+export interface ColorPickerSwatchesProps {
+  onSelectColor: (hex: string) => void
+  currentColor: string
+}
+
+const SWATCH_SIZE = 24
+
+export const ColorPickerSwatches = React.memo((props: ColorPickerSwatchesProps) => {
+  const { currentColor, onSelectColor } = props
+
+  const storeColorSwatches = useEditorState(
+    Substores.restOfEditor,
+    (store) => store.editor.colorSwatches,
+    'ColorPickerPresets color presets',
+  )
+
+  const [editing, setEditing] = React.useState(false)
+  const [colorSwatches, setColorSwatches] = React.useState<Array<ColorSwatch>>(storeColorSwatches)
+
+  const dispatch = useDispatch()
+
+  React.useEffect(() => {
+    dispatch([updateColorSwatches(colorSwatches)], 'everyone')
+  }, [colorSwatches, dispatch])
```

Comment:
interesting. I wonder if we should turn this into a hook that allows for "state with optimistic instant local update + global dispatch one frame later"